code/modules/bitrunning/objects/boulder_pushing.dm

```diff
@@ -0,0 +1,251 @@
+	icon_state = "boulder_checkpoint"
+	density = FALSE
+	anchored = TRUE
+	flags_1 = INDESTRUCTIBLE
+
+
+/obj/bitrunning/boulder_checkpoint/CanAllowThrough(atom/movable/mover, border_dir)
+	if(istype(mover, /obj/bitrunning/hungry_customer))
+		if(border_dir == NORTH)
+			return FALSE
+	. = ..()
+
+
+/obj/bitrunning/customer_destination
+	name = "customer destination"
+	desc = "If you can see this, file a bug report!"
+	icon = 'icons/obj/machines/bitrunning.dmi'
+	base_icon_state = "boulder_destination"
+	icon_state = "boulder_destination"
+	alpha = 0
+	mouse_opacity = 0
+	density = FALSE
+	anchored = TRUE
+	flags_1 = INDESTRUCTIBLE
+	var/victory_line = "Finally! I'll give you a five star review on Lift! Here's a tip!"
+
+/obj/bitrunning/customer_speech
+	name = "customer speech"
+	desc = "If you can see this, file a bug report!"
+	icon = 'icons/obj/machines/bitrunning.dmi'
+	base_icon_state = "customer_speech"
+	icon_state = "customer_speech"
+	alpha = 0
+	mouse_opacity = 0
+	density = FALSE
+	anchored = TRUE
+	flags_1 = INDESTRUCTIBLE
+	var/list/possible_lines = list(
+		"I'm so tired from standing...",
+		"I have chronic back pain, please hurry up and get me a seat!",
+		"I'm not going to tip if I don't get a seat."
+	)
+	var/fired = FALSE
+
+
+/obj/bitrunning/customer_waiter
+	name = "busy waiter"
+	desc = "They look incredibly busy, better not get in their way."
+	icon = 'icons/obj/machines/bitrunning.dmi'
+	base_icon_state = "boulder_blocker"
+	icon_state = "boulder_blocker"
+	density = 1
+	move_force = MOVE_FORCE_EXTREMELY_STRONG
+	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	flags_1 = INDESTRUCTIBLE
+	var/starting_dir = EAST
+	var/current_dir = EAST
+	var/speed = 3
+	var/speech_sound = 'sound/mobs/non-humanoids/tourist/tourist_talk_french.ogg'
+	var/list/bumped_lines = list(
+		"Out of the way!",
+		"Order for table seven!",
+		"Make way!",
+		"Behind!",
+	)
+	var/said_blocked_line = FALSE
+	var/datum/move_loop/loop
+
+/obj/bitrunning/customer_waiter/east
+	starting_dir = EAST
+
+/obj/bitrunning/customer_waiter/east/slow
+	speed = 5
+
+/obj/bitrunning/customer_waiter/east/fast
+	speed = 1
+
+/obj/bitrunning/customer_waiter/west
+	starting_dir = WEST
+
+/obj/bitrunning/customer_waiter/west/slow
+	speed = 5
+
+/obj/bitrunning/customer_waiter/west/fast
+	speed = 1
+
+/obj/bitrunning/customer_waiter/Initialize(mapload)
+	. = ..()
+	current_dir = starting_dir
+	loop = GLOB.move_manager.move(src, starting_dir, delay = speed, flags = MOVEMENT_LOOP_START_FAST|MOVEMENT_LOOP_IGNORE_PRIORITY)
+	RegisterSignal(loop, COMSIG_MOVELOOP_POSTPROCESS, PROC_REF(check_bump))
+
+/obj/bitrunning/customer_waiter/proc/check_bump(datum/move_loop/source, result)
+	SIGNAL_HANDLER
+	if(result == MOVELOOP_FAILURE) // if we can't move, go the other direction
+		var/turf/possible_obstacle = get_step(src, current_dir)
+		var/blocked_by_idiot = FALSE
+		for(var/atom/idiot in possible_obstacle)
+			if(istype(idiot, /obj/bitrunning/hungry_customer))
```

Comment:
I'm not reviwing this all rn but just like, type the for loop there's no reason to split like this 

also early return this outer block